src/ledger/LedgerManagerImpl.cpp

```diff
@@ -696,6 +696,16 @@ during replays.
 void
 LedgerManagerImpl::closeLedger(LedgerCloseData const& ledgerData)
 {
+    // If we do not support ledger version, we can't apply that ledger, fail!
+    if (mCurrentLedger->mHeader.ledgerVersion > getDesiredLedgerVersion())
+    {
+        CLOG(ERROR, "Ledger") << "Unknown ledger version: "
+                              << mCurrentLedger->mHeader.ledgerVersion;
+        throw std::runtime_error(
+            fmt::format("cannot apply ledger with not supported version: {}",
+                        mCurrentLedger->mHeader.ledgerVersion));
+    }
+
```

Comment:
this needs to be moved next to the condition that checks for the hash as there are other variables updated